File: w24w09javalotto/src/main/java/kr/ac/kumoh/s20210463/w24w09javalotto/service/LottoService.java
package kr.ac.kumoh.s20210463.w24w09javalotto.service;

import kr.ac.kumoh.s20210463.w24w09javalotto.model.LottoNumber;
import kr.ac.kumoh.s20210463.w24w09javalotto.repository.LottoRepository;
import org.springframework.stereotype.Service;

import java.util.HashSet;
import java.util.Random;
import java.util.Set;

@Service
public class LottoService {

    private final LottoRepository lottoRepository;
    private final Random random = new Random();

    public LottoService(LottoRepository lottoRepository) {
        this.lottoRepository = lottoRepository;
    }

    public LottoNumber getLuckyNumbers() {
        Set<Integer> numbers = new HashSet<>();

        while (numbers.size() < 6) {
            numbers.add(random.nextInt(1, 46));
        }

        int[] lottoNumbersArray = numbers.stream().sorted().mapToInt(Integer::intValue).toArray();
        LottoNumber lottoNumber = new LottoNumber(lottoNumbersArray);

        return lottoRepository.save(lottoNumber);
    }

}

File: w24w07lotto/src/main/kotlin/kr/ac/kumoh/s20210463/w24w07lotto/service/LottoService.kt
package kr.ac.kumoh.s20210463.w24w07lotto.service

import org.springframework.stereotype.Service
import kotlin.random.Random

@Service
class LottoService {
    fun getLuckyNumbers(): IntArray {
        val numbers = mutableSetOf<Int>()

        while (numbers.size < 6) {
            numbers.add(Random.nextInt(1, 46));
        }

        return numbers.toSortedSet().toIntArray()
    }
}